### PR TITLE
Lifecycle tests for Handle Holders within the AbstractArcHost.

### PR DESCRIPTION
**Pull request by alxrsngrtn**
Also includes typo fix.

This PR is broken out of #6957.

### CODE REVIEW

File: javatests/arcs/core/host/HandleHolderLifecycleTest.kt [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      callbackExecuted = true
+      handleHolder = particle.handles
+      particle
+    }
+
+    // Calls ArcHost.startArc()
+    allocator.startArcForPlan(MultiHandleTestPlan).waitForStart()
+    assertThat(callbackExecuted).isTrue()
+
+    // Handles in handle holder are set
```
Comment (alxrsngrtn) [inline javatests/arcs/core/host/HandleHolderLifecycleTest.kt:194]: Question for the reviewer: Is there a good way to compare types on the handles in the holder vs the plan?

File: javatests/arcs/core/host/HandleHolderLifecycleTest.kt [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  }
+
+  /** A [HandleHolder] with relaxed constraints and method flags used for tests. */
+  class HandleHolderSpy(val parent: HandleHolderBase) : HandleHolderBase("Test", emptyMap()) {
+
+    var detachWasCalled = false
+    var resetWasCalled = false
+
+    init {
+      parent.handles.entries.forEach { (name, handle) -> this.setHandle(name, handle) }
```
Comment (cromwellian) [inline javatests/arcs/core/host/HandleHolderLifecycleTest.kt:132]: won't this be empty on particle construction anyway?

File: javatests/arcs/core/host/HandleHolderLifecycleTest.kt (hunk 1 above)
Comment (cromwellian) [reply javatests/arcs/core/host/HandleHolderLifecycleTest.kt:194]: Unfortunately no. The BaseHandle has a spec object with the type, but this is hidden by the wrappers generated by createHandle(). I suppose you could make a HandleManagerImpl that doesn't wrap in createHandle() and returns the SingletonHandle/CollectionHandle instances directly?

File: javatests/arcs/core/host/HandleHolderLifecycleTest.kt [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    }
+
+    // Calls ArcHost.startArc()
+    allocator.startArcForPlan(MultiHandleTestPlan).waitForStart()
+    assertThat(callbackExecuted).isTrue()
+
+    // Handles in handle holder are set
+    assertThat(handleHolder?.isEmpty()).isFalse()
+    assertThat(handleHolder?.getHandle("data")).isNotNull()
+    assertThat(handleHolder?.getHandle("data")?.mode).isEqualTo(HandleMode.Read)
```
Comment (cromwellian) [inline javatests/arcs/core/host/HandleHolderLifecycleTest.kt:197]: You could generalize this by looping over 
```
MultiHandleTestPlan.particles.forEach { (handleName, handleSpec) -> 
  // test code
}
```

**Review by cromwellian — APPROVED** [review]

File: javatests/arcs/core/host/HandleHolderLifecycleTest.kt (hunk 3 above)
Comment (alxrsngrtn) [reply javatests/arcs/core/host/HandleHolderLifecycleTest.kt]: Thanks!

File: javatests/arcs/core/host/HandleHolderLifecycleTest.kt (hunk 1 above)
Comment (alxrsngrtn) [reply javatests/arcs/core/host/HandleHolderLifecycleTest.kt:190]: I'm satisfied with a no. I was wondering if there was some convention in the codebase. Maybe, we'll figure out a non-kludgy way of testing this later (I imagine this case is covered by unit tests somewhere?).

**Pull request closed without merging**